test/dot_s/basics.s

```diff
@@ -48,7 +48,7 @@ main:                                   # @main
 	end_block
 	i32.const	$push11=, -12
 	i32.add 	$0=, $0, $pop11
-	i32.const	$discard=, main # just take address for testing
+	i32.const	$discard=, main@FUNCTION # just take address for testing
```

Comment:
Or, to be more precise, s2wasm has always just unconditionally stripped it from the names, so it didn't care whether it was there or not. With this PR we use that information to tell whether the relocation should be for a function or an object (instead of looking for an object with that name and assuming it's a function otherwise).